### PR TITLE
added BasePanel Component

### LINKED ISSUE #152: Create `BasePanel` component
### Goal
New reusable `BasePanel` component in the storyook

### Context
Based on the proposal made in https://github.com/sidestream-tech/unified-auctions-ui/issues/145, we want to have a new kind of base element for our UI: a panel. A panel is a component consisting of:
- prop `state`
  - `inactive`
  - `incorrect`
  - `complete`
  - `notice`
- prop and slots for each state
- An icon (based on the state)
- Default child slot to include content within the frame
- An expandable logic _with animation_ (default is based on the state, but the title is still user-clickable/collapsable/expandable and reacts to the `state` change)

Notes: 
- This component doesn't need to have `isExplanationsShown` prop, since this logic will be inside the child slot
- For the `prop and slot` logic, please use "fallback content" feature https://v2.vuejs.org/v2/guide/components-slots.html#Fallback-Content
- Example usage of the component:
```
<BasePanel
    :state="currentState"
    inactive="The VAT balance is unknown until a wallet is connected"
    incorrect="The amount is not present in the VAT"
    complete="The amount is present in the VAT"
>
  Content and buttons
</BasePanel>
```

### Assets
- `inactive`
    MakerDAO@2x (44)

- `incorrect`
    MakerDAO@2x (43)

- `complete`
   MakerDAO@2x (45)

- `notice`
   MakerDAO@2x (47)
 
- Slotted title
    MakerDAO@2x (48)

- Stack of the panels (please note that the panel borders should overlap each other with `.BasePanel + .BasePanel` css selector)
    MakerDAO@2x (49)

### Tasks
- [x] Create a BasePanel component
- [ ] Add stories for all possible use-cases
  - [x] Multi-line title
  - [x] Title via named slot (eg with the wallet address)
  - [ ] Stack of the panels
  - [ ] Animation between `incorrect` and `complete` states

### PR DESCRIPTION
**Pull request by simontree**
closes #152

### CODE REVIEW

File: frontend/.prettierrc [hunk 1]
```diff
@@ -3,5 +3,6 @@
     "semi": true,
     "singleQuote": true,
     "arrowParens": "avoid",
-    "printWidth": 119
+    "printWidth": 119,
```
Comment (benediktkaiser) [inline frontend/.prettierrc:6]: Why did you do this?

**Review by benediktkaiser — CHANGES_REQUESTED** [review]

File: frontend/components/common/BasePanel.vue [hunk 2]
```diff
@@ -0,0 +1,69 @@
+<template>
+    <modal :visible="true" :mask-closable="false" :closable="false" :footer="null">
```
Comment (valiafetisov) [inline frontend/components/common/BasePanel.vue:2]: Why did you use a modal? The issue and the mock doesn't say anything about the modal 🙂

File: frontend/components/common/BasePanel.vue [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            </div>
+            <div class="border-solid border-2 p-5 gap-5">
+                <div class="grid grid-cols-2">
+                    <span>Amount in wallet:</span>
+                    <span>100 DAI</span>
+                </div>
+                <div>lorem ipsum</div>
+                <div class="flex justify-end p-5 gap-5">
+                    <BaseButton type="primary">Click here</BaseButton>
+                </div>
```
Comment (valiafetisov) [inline frontend/components/common/BasePanel.vue:26]: This should not be part of the `BasePanel`, but should be provided as content (in other words as a `<slot />`)

File: frontend/components/common/BasePanel.vue [hunk 4]
```diff
@@ -0,0 +1,69 @@
+<template>
+    <modal :visible="true" :mask-closable="false" :closable="false" :footer="null">
+        <!-- Default Base Panel -->
+        <div v-if="state === 'default'">
```
Comment (valiafetisov) [inline frontend/components/common/BasePanel.vue:4]: What is the `default` state? The issue doesn't mention it

**Review by valiafetisov — COMMENTED** [review]
- [ ] please add icons
- [ ] please add animation

**Comment by benediktkaiser** [comment]
Closed by accident. Reopening this PR.

**Pull request closed without merging**